examples/s3/v1beta1/directorybucket.yaml

```diff
@@ -0,0 +1,15 @@
+# https://docs.aws.amazon.com/AmazonS3/latest/userguide/s3-express-Endpoints.html
+apiVersion: s3.aws.upbound.io/v1beta1
+kind: DirectoryBucket
+metadata:
+  annotations:
+    meta.upbound.io/example-id: s3/v1beta1/directorybucket
+    crossplane.io/external-name: ${Rand.RFC1123Subdomain}--usw2-az1--x-s3
+  labels:
+    testing.upbound.io/example-name: example
+  name: example
+spec:
+  forProvider:
+    location:
+    - name: usw2-az1
```

Comment:
This looks like a singleton list, which I'm surprised to see here. Perhaps the code which converts it into an embedded object only supports SDK resources, not framework?

Especially since this parameter should go into the external name config, I think we should do the necessary work to make this into an embedded object.

Reply:
@mbbush, `location` [is indeed a singleton list](https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/s3/directory_bucket.go#L99-L124). Our singleton list conversion implementation processes only Plugin SDK resources. Because Plugin Framework resource schema has a different structure (and type), singleton list detection and conversion should be implemented separately for it.

If I recall correctly though, singleton list fields shouldn't exist in Plugin Framework resources, in the first place, because Plugin Framework schema is able to represent them as embedded objects. Unfortunately, I can't remember why and how at the moment.

Reply:
A clarification… We [traverse schemas of all resources](https://github.com/crossplane/upjet/blob/main/pkg/config/provider.go#L376), Plugin Framework and Plugin SDKv2, but our [singleton list detector](https://github.com/crossplane/upjet/blob/main/pkg/config/schema_conversions.go#L66-L77) doesn't detect singleton lists of Plugin Framework schemas.

Reply:
We need to make changes on the upjet side to convert the singleton list to embedded object in the framework resources. I've created an issue for [this](https://github.com/crossplane/upjet/issues/430) on upjet, we can follow up from there.